<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="markerId1" default="582" />
  <arg name="markerId2" default="62" />
  <arg name="markerSize" default="0.034" /> <!-- in m -->

  <node pkg="aruco_ros" type="double" name="aruco_double">
    <remap from="/camera_info" to="/xtion/rgb/camera_info" />
    <remap from="/image" to="/xtion/rgb/image_rect_color" />
    <param name="image_is_rectified" value="True" />
    <param name="marker_size" value="$(arg markerSize)" />
    <param name="marker_id1" value="$(arg markerId1)" />
    <param name="marker_id2" value="$(arg markerId2)" />
    <param name="parent_name" value="/base_footprint" />
    <param name="child_name1" value="/box1" />
    <param name="child_name2" value="/box2" />
    <param name="reference_frame" value="/base_footprint" /> <!-- frame in which the marker pose will be expressed -->
    <param name="camera_frame" value="xtion_rgb_optical_frame" />
    <param name="marker_frame" value="aruco_frame" />
    <param name="corner_refinement" value="SUBPIX" />
  </node>
</launch>
